src/dbnode/client/config.go

```diff
@@ -335,3 +341,29 @@ func (c Configuration) NewAdminClient(
 
 	return NewAdminClient(opts)
 }
+
+func loadSchemaRegistryFromKVStore(schemaReg namespace.SchemaRegistry, kvStore kv.Store) error {
+	if kvStore == nil {
+		return errors.New("m3db metadata store is not configured properly")
+	}
+	as := kvadmin.NewAdminService(kvStore, "", nil)
+	nsReg, err := as.GetAll()
+	if err != nil {
+		return xerrors.Wrap(err, "could not get metadata from metadata store")
+	}
+	nsMap, err := namespace.FromProto(*nsReg)
+	if err != nil {
+		return xerrors.Wrap(err, "could not unmarshall metadata")
+	}
+	merr := xerrors.NewMultiError()
+	for _, metadata := range nsMap.Metadatas() {
+		err = schemaReg.SetSchemaHistory(metadata.ID(), metadata.Options().SchemaHistory())
+		if err != nil {
+			merr.Add(xerrors.Wrapf(err, "could not set schema history for namespace %s", metadata.ID().String()))
+		}
+	}
+	if merr.Empty() {
```

Comment:
I think you can just `return merr.FinalError()`

Reply:
cool